### ISSUE TITLE
GetValueArray should return an empty array when the setting value is {}

### ISSUE THREAD

**Issue by asmolero** [cemdervis/SharpConfig#16]
And its ArraySize property should be 0 instead of -1.

Currently the method GetValueArray<string> throws a System.OverflowException.

**Comment by cemdervis** [comment]
Fixed with the latest commit! New version should be up soon.

**Comment by asmolero** [comment]
Thank you for fixing this issue! Maybe you have already considered this approach and discarded it for some reason, but wouldn't it make more sense to use something based in string.Split instead of manually finding the delimiters and dealing with them at the character level?

**Comment by cemdervis** [comment]
Hey, you're welcome! I considered string.Split(), but discarded it because of performance, although it should not be a huge difference. The main reason why I discarded it was because of flexibility. You know, I could've used string.Split(...).Length in the ArraySize property, but that would be a waste, because I don't actually need the string values that string.Split() returns for me. I'm just interested in the number of elements.
For GetValueArray(), I decided to implement a custom iterator (SettingArrayEnumerator) to avoid creating a string array (that string.Split() would create), as I'm just interested in every single element.
All in all, I think you're right, as the performance difference should be minimal-to-none when using string.Split(), because the string array would just be an array of (string) references.

In the end, I chose flexibility and control over simplicity, to be able to adapt to new features (e.g. array syntax). As you said, string.Split() should also work.
